File: TravelPlannerApp/SharingService/Models/ShareToken.cs
using SharingService.Enums;

namespace SharingService.Models
{
    public class ShareToken
    {
        public int Id { get; set; }
        public string Token { get; set; }
        public int TripId { get; set; }
        public ShareAccessType AccessType { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime ExpiresAt { get; set; }
        public bool IsActive { get; set; } = true;
    }
}

File: frontend/src/models/ShareToken.js
export class ShareToken {
  constructor(id, token, tripId, accessType, expiresAt, isActive) {
    this.id = id
    this.token = token
    this.tripId = tripId
    this.accessType = accessType
    this.expiresAt = expiresAt
    this.isActive = isActive
  }

  isViewOnly() {
    return this.accessType === 0
  }

  isExpired() {
    return new Date(this.expiresAt) < new Date()
  }
}